articles/converting-html-content-to-pdf-with-pdfkit-for-python/index.md

```diff
@@ -0,0 +1,177 @@
+
+### **Pdfmake**
+
+**[Pdfmake](http://pdfmake.org/#/)** is an open-source library for converting web documents to PDF, it uses paragraphs, columns, lists, tables, canvas, etc... Also allows users to declare their styles, use custom fonts, build a DSL and extend the framework.
+
+However, the pdfmake converts the document from scratch rather than from the existing HTML text, *i.e.* the user providers data for the header, content, footer, etc. This makes the configuration of pdfmake more difficult compared to other libraries.
+
+### **Pypdf**
+
+Asides pdfmake, there are other libraries such as **[pyPdf](https://pypi.org/project/pyPdf/)**. Pypdf is a Pure-Python library built as a PDF toolkit capable of splitting, merging, cropping, and transforming the pages of PDF files. It can also add custom data, viewing options, and passwords to PDF files. It can retrieve text and metadata from PDFs as well as merge entire files together. However, with pypdf text might not be ordered logically, it is bested used for simpler and small documents as the document becomes larger, text ordering might become complex.
+
+### **Pdfcrowd**
+
+Pdfcrowd is an open-source API that is capable of converting PDF from dynamic and static HTML in various environments. It also generates PDFs and screenshots from web pages and HTML documents. However, pdfcrowd unlike other libraries is not completely free it comes with a paid browser license to unlock a series of advanced options, like controlling the page size, password-protecting the PDF, and preventing printing and copying.
+
+**Pdfminer**, **PyMuPDF**, etc are also libraries for converting HTML to PDF in python. However, in this article, we will be using **pdfkit**. Pdfkit is one of the better approaches as, it renders HTML into PDF with various image formats, HTML forms, and other complex printable documents.
+
+**N.B:** There are also libraries and tools for converting PDF in Django and JavaScript.
+
+### **What is pdfkit?**
+
+Pdfkit is an open-source library that generates HTML from web URLs, local files a string and converts this text to PDF format. Web apps often contain tables, charts, and graphs asides text. Export these data professional, PDF converters are the best options, and pdfit converts data regardless of the format, either image, table, graphs, HTML forms, and other complex printable documents.
+
+We will be using wkhtmltopdf alongside pdfkit to achieve perfect work. Wkhtmltopdf is an open-source command-line tool that renders HTML into PDF and various image formats. To know more about wkhtmltopdf see [https://wkhtmltopdf.org/](https://wkhtmltopdf.org/).
+
+First thing first, let's begin by installing pdfkit.
+
+### **Installing pdfkit**
+
+Pdfkit is installed using the pip command in your terminal.
+
+```bash
+pip install pdfkit
+
+```
+
+![pdfkit-installation](/engineering-education/python-projects-for-beginners/pdfkit.png)

```

Comment:
What does this image do, tell the user...., for instance, you can say .."as shown in the image below to explain your concept further..."